clinica/utils/filemanip.py

```diff
@@ -457,15 +462,98 @@ def extract_metadata_from_json(json_file: str, list_keys: List[str]) -> List[str
             f"Clinica could not open the following JSON file: {json_file}"
         )
     list_values = []
-    missing_keys = []
+    data_keys = []
     for key in list_keys:
         if key in data:
             list_values.append(data[key])
-        else:
-            missing_keys.append(key)
-    if len(missing_keys) > 0:
-        error_message = f"Clinica could not find the following keys in the following JSON file: {json_file}:\n- "
-        error_message += "\n- ".join(missing_keys)
-        raise ClinicaException(error_message)
-
-    return list_values
+            data_keys.append(key)
+        # else:
+        #     missing_keys.append(key)
+    missing_keys = set(list_keys).difference(set(data_keys))
+    if len(missing_keys) > 0 and handle_missing_keys is None:
+        raise ClinicaException(
+            f"Clinica could not find the following keys in the following JSON file: {missing_keys}."
+        )
+    if handle_missing_keys:
+        patch = handle_missing_keys(data, missing_keys)
+        data.update(patch)
+    return [data[k] for k in list_keys]
+
+
+def handle_missing_keys_dwi(data: dict, missing_keys: set) -> dict:
+    """Find alternative fields from JSON file to replace those which were not found.
```

Comment:
Maybe you could be more specific about this JSON file. When reading this I don't really know which file we are talking about. I know it's related to DWI somehow because of the function name, but that's it. I think providing an example name of this JSON file could be useful. WDYT ?

Reply:
Done !